### ISSUE TITLE
Configurable UI

### ISSUE THREAD

**Issue by IanMayo** [DeepBlueCLtd/narr-next#16]
When loading, we should read the UI definition from a file, then produce categories and events to match that file.

Sample UI file here: https://github.com/DeepBlueCLtd/narr-next/blob/master/schema/example.js

Note: for UI generation, the fields within an entry will not be shown on the screen - they will be used when we generate the input dialogs.  So, they should be stored within the UI component, ready for when the button has been clicked.

**Comment by Morphexe** [comment]
@IanMayo Completed, I also took the liberty to add the modal dialogs as well. :) 

 Currently, they only handle the default Html field types (text,text-area, number, etc), it was provided by reacstrap by default,  but I intend to be able to add custom fields, like enums, date pickers, if/when we required.

I also took the liberty of writing  Tests for the simple UI generation, so I think I got most of the functionality covered - but they are very simple, and they can serve as a start point for when someone else needs to write a test for any part of the application. I installed Enzyme to help test React Components.
